### ISSUE TITLE
reproducibility issues

### ISSUE THREAD

**Issue by zhch-sun** [NVIDIA/DALI#729]
Hi guys, thank you for the awesome work! 
I have two things about the reproducibility about Dali. 
1. Is it possible to set seed for all the operations? Currently only the class Pipeline support add a seed, and this seed is not the seed for operations like HostDecoderRandomCrop and CropMirrorNormalize. Would you like to add an API for setting seed to these operations? 
2. I guess there is a bug in the recordIO reader.
I train a model(e.g. Res18) in distributed training for an epoch and evaluate the model and save the model. And then load the saved model and evaluate: the results are different from before! And part of the reason is BN, after using sync-bn, all the testing results are exactly the same except for the first batch.
So I print the data of first batch and I found in the evaluation in training one batch on a card are all zero. While in the testing with loaded models this does not happen often, although it happens sometimes.

**Comment by zhch-sun** [comment]
By the way, I use Pytorch training script, together with RecordIO reader.

**Comment by JanuszL** [comment]
Hi,
With 0.8 release all operators allows you to set seed for it.
Regarding the problem you see - can you provide some reproduction script for that? Does this ahppen on all datasets, or just only training/evaluation?

**Comment by zhch-sun** [comment]
Thank you for the quick response. Now I only test Imagenet dataset.

**Comment by zhch-sun** [comment]
I will try to provide a script for reproduction.

**Comment by zhch-sun** [comment]
I did some other experiments and Now I guess the bug might be related to some hardware bug. Same parameters sometimes lead to reproducible results, sometimes not. I think you may just close this issue. Thank you @JanuszL
